tests/integration/cattletest/core/test_agent.py

```diff
@@ -4,7 +4,7 @@
 
 # sim_context is included to ensure that the simulator context creates
 # the first agent and thus the first external simulator pool
-def test_agent_create(admin_client, sim_context):
+def test_agent_create(admin_client, internal_test_client, sim_context):
```

Comment:
I realize that you did not do this one, but do you want to fix this instance of client = admin_client and just make the test explicitly use admin_client so that it is more clear?